src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1567,15 +1620,49 @@ SetWorkerColumn(WorkerNode *workerNode, int columnIndex, Datum value)
 	Datum values[Natts_pg_dist_node];
 	bool isnull[Natts_pg_dist_node];
 	bool replace[Natts_pg_dist_node];
-	char *metadataSyncCommand = NULL;
 
+	if (heapTuple == NULL)
+	{
+		ereport(ERROR, (errmsg("could not find valid entry for node \"%s:%d\"",
+							   workerNode->workerName, workerNode->workerPort)));
+	}
+
+	memset(replace, 0, sizeof(replace));
+	values[columnIndex - 1] = value;
+	isnull[columnIndex - 1] = false;
+	replace[columnIndex - 1] = true;
+
+	heapTuple = heap_modify_tuple(heapTuple, tupleDescriptor, values, isnull, replace);
+
+	CatalogTupleUpdate(pgDistNode, &heapTuple->t_self, heapTuple);
+
+	CitusInvalidateRelcacheByRelid(DistNodeRelationId());
+	CommandCounterIncrement();
+
+	WorkerNode *newWorkerNode = TupleToWorkerNode(tupleDescriptor, heapTuple);
+
+	table_close(pgDistNode, NoLock);
+
+	return newWorkerNode;
+}
+
+
+/*
+ * SetWorkerColumnInternal checks if the given workerNode and value is valid or not.
+ * Then it returns the necessary metadata sync command as a string.
+ */
+static char *
+SetWorkerColumnInternal(WorkerNode *workerNode, int columnIndex, Datum value)
```

Comment:
Maybe call it as `GetMetadataSyncCommand` or such? Because at this point it is not an `Set` ting anything, it is just generating a command?